File: Board.cpp
/*
 * board.cpp
 *
 *  Created on: Nov 5, 2017
 *      Author: jonathan
 */

#include "Board.h"

#include <iostream>
using namespace std;

Board::Board(int l , int w) {
	this->length_ = l;
	this->width_ = w;
	this->arr_= new char * [l];
	for (int i = 0; i < l; i ++) {
		this->arr_[i] = new char [w];
		for (int j = 0; j < w; j ++) {
			this->arr_[i][j] = ' ';
		}
	}
	this->arr_[3][3] = 'O';
	this->arr_[4][4] = 'O';
	this->arr_[3][4] = 'X';
	this->arr_[4][3] = 'X';
}

Board:: Board(const Board &b) {
	this->length_ = b.length_;
	this->width_ = b.width_;
	this->arr_= new char * [b.length_];
	for (int i = 0; i < b.length_; i ++) {
		this->arr_[i] = new char [b.width_];
		for (int j = 0; j < b.width_; j ++) {
			this->arr_[i][j] = b.arr_[i][j];
		}
	}
}

Board& Board:: operator= (const Board& b) {
	if (this != &b) {
		for (int i = 0; i < this->getLength(); i ++) {
			delete this->arr_[i];
		}
		delete this->arr_;
		this->arr_= new char * [b.length_];
		for (int i = 0; i < this->getLength(); i ++) {
			this->arr_[i] = new char [b.width_];
			for (int j = 0; j < b.width_; j ++) {
				this->arr_[i][j] = b.arr_[i][j];
			}
		}
	}
	return *this;
}

bool Board:: isNotFull() {
	for (int i = 0; i < this->getLength(); i ++) {
		for (int j = 0; j < this->getWidth(); j ++) {
			if (this->arr_[i][j] == ' ') {
				return true;
			}
		}
	}
	return false;
}

char Board::getCell (int i ,  int j) {
	return this->arr_[i][j];
}

/**
 * before setting a cell, it verifies that parameters of i,j are inside array.
 */
void Board:: setCell(int i , int j , char c) {
	if (i >= 0 and i < this->getLength() and j >= 0 and j< this->getWidth() ) {
		this->arr_[i][j] = c;
	}
}

int Board:: getLength() {
	return this->length_;
}

int Board:: getWidth() {
	return this->width_;
}

Board::~Board() {
	// TODO Auto-generated destructor stub
	for (int i = 0; i < this->length_; i ++) {
		delete [] this->arr_[i];
	}
	delete [] this->arr_;
}

File: java for ex6/src/game/Board.java
package game;
import java.util.ArrayList;
public class Board {
	// private  ArrayList<ArrayList<Character>> b;
	private char [][]b;
	private int length;
	private int width;

	public Board(int length , int width) {
		this.length = length;
		this.width = width;
		/*
		this.b = new ArrayList<ArrayList<Character>>(length);
		for (int i = 0; i < length; i ++) {
			this.b.add(new ArrayList<Character>(width));
			ArrayList temp = this.b.get(i);
			for (int j = 0; j < width; j ++) {
				temp.set(j , ' ');
			}
		}
		this.b.get(3).set(3 , 'X');
		*/
		this.b = new char[length][width];
		for (int i = 0; i < length; i ++) {
			for (int j = 0; j < width; j ++) {
				b[i][j] = ' ';
			}
		}
		this.b[3][3] = 'O';
		this.b[4][4] = 'O';
		this.b[3][4] = 'X';
		this.b[4][3] = 'X';
	}

	public char getCell(int i , int j) {
		//return this.b.get(i).get(j);
		return this.b[i][j];
	}

	public void setCell(int i , int j , char c) {
		//this.b.get(i).set(i , c);
		this.b[i][j] = c;
	}

	public int  getLength() {
		return this.length;
	}

	public int getWidth() {
		return this.width;
	}

	public boolean isNotFull() {
		for (int i = 0; i < this.length; i ++) {
			for (int j = 0; j < this.width; j ++) {
				if (this.b[i][j] == ' ') {
					return true;
				}
			}
		}
		return false;
}


	public static void main (String[] args) {
		Board b = new Board(8 ,8);
		System.out.println(b.getCell(3, 3));
	}
}